### ISSUE TITLE
Closing a tab sometimes closes Godot

### ISSUE THREAD

**Issue by YeldhamDev** [godotengine/godot#10409]
**Ubuntu 17.04 - Godot version: Since 3.0.alpha1**

**Issue description:**
It is uncommon, but sometimes, when closing a tab, the whole program closes completely.

**Comment by eon-s** [comment]
What kind of tab, 2D, 3D, script tab, any? 

Do you have a plugin/tool running, some specific nodes in scenes or selection on the inspector when crashes?

**Comment by YeldhamDev** [comment]
Scene tabs, no plugins or tools, and I don't remember the scenes having something in common when it happens.

**Comment by toger5** [comment]
Would be really helpful to figure out some kind of paddern? (i also think godot does not close but just crashes...)
Issue never occured to me, but i will keep an eye open.

**Comment by YeldhamDev** [comment]
The problem is that it doesn't seem to have a pattern to begin with. It already happened with both single and multiple tabs, with various types of contents.

And it doesn't feel like a crash, as it just closes silently.

**Comment by toger5** [comment]
Interesting. Do you have the chance to run it in debug mode?

**Comment by YeldhamDev** [comment]
What do you mean?

**Comment by toger5** [comment]
If you can compile it with debug symbols and see if you can get additional information from your debug tool?
If not: does the consol which opens with godot close too? What does it say if it doesnt close...

**Comment by YeldhamDev** [comment]
I don't compile it myself, I grab the builds compiled by calinou [here](https://bintray.com/calinou/godot/editor).

No console opens with Godot when I use it on my Ubuntu laptop.

**Comment by sketchyfun** [comment]
Open a terminal window and start godot via that, any errors should appear there

**Comment by eon-s** [comment]
If it is a segfault, there may not be any output.

**Comment by YeldhamDev** [comment]
Ok, I got good and bad news:

**Good news:** I got to replicate the crash while having the terminal, and I think it's not about closing the tabs, but closing the tabs' scripts.

**Bad news:** In the process of copy and pasting the crash dump, my computer decided it was time to fucking die, and completely froze while having the right-click menu in the middle of the crash dump.

```
program crashed with signal 11
[1] /lib/x86_64-linux-gnu/lib [Part blocked by the right-click menu] 7f0f40a757f0] (??:0)
[2] [0xeab2e70] (??:0)
```

**Comment by akien-mga** [comment]
Can you describe steps to reproduce the bug? Just by closing script tabs I don't get a crash in my tests.

**Comment by YeldhamDev** [comment]
@akien-mga The problem is that it seems to happen when it wants to. So far, this is the only pattern: Triggering something that closes scripts in the Script Editor (e. g. closing a scene tab).

It tends to happen more often in those situations:

- Multiple scripts being closed at the same time.
- Closing scripts that have been removed from a node.

**Comment by mhilbrunner** [comment]
This seems to be a duplicate of https://github.com/godotengine/godot/issues/10359.

**Comment by akien-mga** [comment]
> This seems to be a duplicate of #10359.

Related for sure, but it's hard to say if it's a duplicate or not based on the information we have so far.

**Comment by akien-mga** [comment]
Can you still reproduce this issue @YeldhamDev? Ideally with a self-compiled debug binary, as so far we never had a full stacktrace.

**Comment by volzhs** [comment]
I had this issue within 10 days but it's random and can't figure out steps so far.

**Comment by YeldhamDev** [comment]
@akien-mga Basically what @volzhs said. It's very random.

**Comment by akien-mga** [comment]
#10359 was just fixed by #15498, and since both issues were related, would be nice if you could confirm this crash here *yet* another time :D

**Comment by YeldhamDev** [comment]
@akien-mga I test it for a bit, and doesn't seems to be happening for now, so I guess I can close it. I will warn you if the problem ever arises again.
